joern-cli/frontends/c2cpg/src/main/scala/io/joern/c2cpg/parser/JSONCompilationDatabaseParser.scala

```diff
@@ -30,18 +30,18 @@ object JSONCompilationDatabaseParser {
     */
   private val includeInCommandPattern = """-I(\S+)""".r
 
-  def parse(compileCommandsJson: String): mutable.LinkedHashSet[CommandObject] = {
+  def parse(compileCommandsJson: String): Option[CompilationDatabase] = {
     try {
       val jsonContent       = IOUtils.readEntireFile(Paths.get(compileCommandsJson))
       val json              = ujson.read(jsonContent)
       val allCommandObjects = mutable.LinkedHashSet.from(json.arr)
-      allCommandObjects.map { obj =>
+      Some(CompilationDatabase(allCommandObjects.map { obj =>
         CommandObject(obj("directory").str, safeArguments(obj), safeCommand(obj), obj("file").str)
-      }
+      }))
     } catch {
       case t: Throwable =>
```

Comment:
```suggestion
      case NonFatal(e) =>
```

and import `scala.util.control.NonFatal`

To be sure not to catch things like OOMError